### PR TITLE
PBL-21952 PBL-24920 Add ability to clear log history

### PR DESCRIPTION
**Pull request by Spacerat**
This adds a clear logs button to the app logs window, fixed in the bottom right hand corner of the screen. It also adds Clear App Logs to the (currently disabled) cmd-shift-P menu.

### CODE REVIEW

**Comment by Katharine** [comment]
Screenshot?

**Comment by Spacerat** [comment]
https://i.imgur.com/bAoVtfv.png

After making this I remembered that the Timeline Preview window has larger orange buttons in the bottom right corner, so if you prefer I could style it like that instead. (or keep it small but make it orange)

**Comment by Spacerat** [comment]
new screenshot 

This is how it would look with the same style as the timeline page

**Comment by fletchto99** [comment]
What about having a download logs button beside the clear logs button? And then instead of having it overtop of the logs, put them both below the logs?

**Comment by Spacerat** [comment]
Well, the advantage of having it overlay the logs is that, when the browser window is wider (and it normally is), it never actually covers the logs since the lines aren't long enough. Having the buttons below the logs would take up valuable log space (well, about three lines, but still...).

A download logs button - could do. Or a select-all-logs button? Do you think you'd use either?

**Comment by Katharine** [comment]
I would prefer to see a button under the view, rather than overlapping it. If it is overlapping, it should be at the top right so it doesn't overrun the latest logs.

If overlapping, use a small button; else use a big one.

(I'd pass on the "save logs" button entirely for now.)

**Comment by Spacerat** [comment]
screen shot 2015-07-29 at 15 24 15
So I gave this more of a thought: people seem to be asking for this feature because they end up with too many logs cluttering everything up, but really a button to delete all the logs is a compromise solution since you end up binning all the nice logs you already have.

In the screenshot above I've implemented an "archive" button* in the bar on the right. When you click it, all current logs are folded away into an archive line, which can then shown or hidden by clicking it. The clear logs button above it deletes all logs and all archives.

* I've used the save icon there but I'll find something more appropriate if we keep this feature.

File: ide/static/ide/js/compile.js [hunk 1]
```diff
@@ -516,6 +516,38 @@ CloudPebble.Compile = (function() {
         });
     };
 
+    var show_clear_logs_prompt = function() {
+        CloudPebble.Prompts.Confirm(gettext("Clear all app logs?"), gettext("This cannot be undone."), function() {
+            mLogHolder.empty();
+            mPreviousDisplayLogs = [];
+            mArchivePoints = [];
+            ga('send', 'event', 'logs', 'delete');
```
Comment (Katharine) [inline ide/static/ide/js/compile.js]: We don't use Google Analytics (much) any more; use the keen system.

Comment (Spacerat) [inline ide/static/ide/js/compile.js]: As far as I can tell, keen is a server side thing only? This doesn't interact with the server.

**Comment by Spacerat** [comment]
I've removed the archive logs button from this PR for now since it was unfinished and just delaying the delete logs button. It should be ready to merge now.

**Comment by Spacerat** [comment]
And I added a save logs button back in! But this time it just downloads the current logs as a text file ( @fletchto99 's idea won after all.)

**Pull request closed without merging**